models/RepPoints/builder.py

```diff
@@ -0,0 +1,722 @@
+                max_index_i = X.reshape(
+                    mx.symbol.slice_axis(max_index_, axis=0, begin=i, end=i + 1), (-1,)
+                )
+                scores_i = X.reshape(
+                    mx.symbol.slice_axis(scores_, axis=0, begin=i, end=i + 1), (-3, -2)
+                )
+                points_i = X.reshape(points_, (-3, -2))
+                preds_refines_i = X.reshape(
+                    mx.symbol.slice_axis(preds_refines_, axis=0, begin=i, end=i + 1), (-3, -2)
+                )
+                scores_i = mx.symbol.take(scores_i, max_index_i)
+                points_i = mx.symbol.take(points_i, max_index_i)
+                preds_refines_i = mx.symbol.take(preds_refines_i, max_index_i)
+                points_i = mx.symbol.slice_axis(points_i, axis=-1, begin=0, end=2)
+                points_xyxy_i = X.concat(
+                    [points_i, points_i], axis=-1, name="points_xyxy_b{}_s{}".format(i, s)
+                )
+                bboxes_i = preds_refines_i * s + points_xyxy_i
+                im_info_i = mx.symbol.slice_axis(im_info, axis=0, begin=i, end=i + 1)
+                h_i, w_i, _ = mx.symbol.split(im_info_i, num_outputs=3, axis=1)
+                l_i, t_i, r_i, b_i = mx.symbol.split(bboxes_i, num_outputs=4, axis=1)
+                clip_l_i = mx.symbol.maximum(mx.symbol.broadcast_minimum(l_i, w_i - 1.0), 0.0)
+                clip_t_i = mx.symbol.maximum(mx.symbol.broadcast_minimum(t_i, h_i - 1.0), 0.0)
+                clip_r_i = mx.symbol.maximum(mx.symbol.broadcast_minimum(r_i, w_i - 1.0), 0.0)
+                clip_b_i = mx.symbol.maximum(mx.symbol.broadcast_minimum(b_i, h_i - 1.0), 0.0)
+                clip_bboxes_i = X.concat(
+                    [clip_l_i, clip_t_i, clip_r_i, clip_b_i],
+                    axis=1,
+                    name="clip_bboxes_b{}_s{}".format(i, s)
+                )
+                scores_dict["img%s" % i] = scores_i
+                bboxes_dict["img%s" % i] = clip_bboxes_i
+            cls_score_ = mx.symbol.stack(
+                *[scores_dict["img%s" % i] for i in range(batch_image)], axis=0
+            )
+            pad_zeros_ = mx.symbol.zeros_like(
+                mx.symbol.slice_axis(cls_score_, axis=-1, begin=0, end=1)
+            )
+            cls_score_ = X.concat([pad_zeros_, cls_score_], axis=-1, name="cls_score_s{}".format(s))
+            bboxes_ = mx.symbol.stack(
+                *[bboxes_dict["img%s" % i] for i in range(batch_image)], axis=0
+            )
+            cls_score_dict["stride%s" % s] = cls_score_
+            bbox_xyxy_dict["stride%s" % s] = bboxes_
+
+        cls_score = X.concat(
+            [cls_score_dict["stride%s" % s] for s in stride], axis=1, name="cls_score_concat"
+        )
+        bbox_xyxy = X.concat(
+            [bbox_xyxy_dict["stride%s" % s] for s in stride], axis=1, name="bbox_xyxy_concat"
+        )
+
+        return cls_score, bbox_xyxy
+
+
+class RepPointsNeck(object):
```

Comment:
It seems this is the same as `RetinaNetNeck` in the `models/retinanet/builder.py`. Consider reusing existing codes.

Reply:
Not exactly the same. `RetinaNetNeck` does not consider normalization, but `RepPointsNeck ` uses GroupNorm and has some differences with `RetinaNetNeckWithBN`. I think it is a trick widely used in mmdetection.